tests/Aspire.Hosting.Tests/Python/AddPythonProjectTests.cs

```diff
@@ -0,0 +1,296 @@
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var (projectDirectory, pythonExecutable, scriptName) = CreateTempPythonProject();
+
+        builder.AddPythonProject("pythonProject", projectDirectory, scriptName, scriptArgs: "test");
+
+        var app = builder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        var executableResources = appModel.GetExecutableResources();
+
+        var pythonProjectResource = Assert.Single(executableResources);
+
+        Assert.Equal("pythonProject", pythonProjectResource.Name);
+        Assert.Equal(projectDirectory, pythonProjectResource.WorkingDirectory);
+
+        if (OperatingSystem.IsWindows())
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "Scripts", "python.exe"), pythonProjectResource.Command);
+        }
+        else
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "bin", "python"), pythonProjectResource.Command);
+        }
+
+        var commandArguments = await ArgumentEvaluator.GetArgumentListAsync(pythonProjectResource);
+
+        Assert.Equal(scriptName, commandArguments[0]);
+        Assert.Equal("test", commandArguments[1]);
+
+        // If we don't throw, clean up the directories.
+        Directory.Delete(projectDirectory, true);
+    }
+
+    private static (string projectDirectory, string pythonExecutable, string scriptName) CreateTempPythonProject(bool instrument = false)
+    {
+        var projectDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(projectDirectory);
+
+        if (instrument)
+        {
+            PreparePythonProject(projectDirectory, PythonApp, InstrumentedPythonAppRequirements);
+        }
+        else
+        {
+            PreparePythonProject(projectDirectory, PythonApp);
+        }
+
+        var pythonExecutable = Path.Combine(projectDirectory,
+            ".venv", "Scripts", OperatingSystem.IsWindows() ? "python.exe" : "python"
+            );
+
+        return (projectDirectory, pythonExecutable, "main.py");
+    }
+
+    private static void PreparePythonProject(string projectDirectory, string scriptContent, string? requirementsContent = null)
+    {
+        var scriptPath = Path.Combine(projectDirectory, "main.py");
+        File.WriteAllText(scriptPath, scriptContent);
+
+        var requirementsPath = Path.Combine(projectDirectory, "requirements.txt");
+        File.WriteAllText(requirementsPath, requirementsContent);
+
+        var prepareVirtualEnvironmentStartInfo = new ProcessStartInfo()
+        {
+            FileName = "python",
+            Arguments = $"-m venv .venv",
+            WorkingDirectory = projectDirectory
+        };
+
+        var createVirtualEnvironmentProcess = Process.Start(prepareVirtualEnvironmentStartInfo);
+        var createVirtualEnvironmentProcessResult = createVirtualEnvironmentProcess!.WaitForExit(10000);
```

Comment:
It would be very useful to capture the output too, incase the command fails, especially on CI.